File: iosApp/iosAppE2ETests/Suites/GuestJourneyTest.swift
import XCTest

/// Guest user journey — browse the app without authenticating.
/// Verifies core screens render, tab navigation works, and
/// auth-gated features redirect to login.
final class GuestJourneyTest: XCTestCase {

    private let robot = AppRobot()

    override func setUp() {
        super.setUp()
        continueAfterFailure = false
        robot.launchApp()
    }

    func testHomeScreenRendersOnLaunch() {
        robot.assertElementDisplayed("HomeUserName")
        robot.assertElementDisplayed("HomeHeroBanner")
    }

    func testBrowseAllTabs() {
        // Home is default
        robot.assertElementDisplayed("HomeUserName")

        // Order tab
        robot.tapTab("ORDER")
        robot.assertElementDisplayed("OrderFeaturedItemsSection")

        // Rewards tab
        robot.tapTab("REWARDS")
        robot.assertElementDisplayed("RewardsPointsSection")

        // Scan tab
        robot.tapTab("SCAN")
        robot.assertElementDisplayed("ScanMemberCard")

        // More tab
        robot.tapTab("MORE")
        robot.assertElementDisplayed("MoreMenuList")

        // Back to home
        robot.tapTab("HOME")
        robot.assertElementDisplayed("HomeUserName")
    }

    func testProfileNavigationRedirectsToLoginWhenUnauthenticated() {
        robot.tapTab("MORE")
        robot.assertElementDisplayed("MoreProfileSection")
        robot.tapElement("MoreProfileSection")

        // Auth interception should redirect to login
        robot.assertElementDisplayed("LoginBranding")
        robot.assertElementDisplayed("LoginSignInButton")
    }

    func testSignInFlowShowsWelcomeScreenThenCompletes() {
        robot.tapTab("MORE")
        robot.tapElement("MoreProfileSection")

        // Login screen appears via auth interception
        robot.assertElementDisplayed("LoginSignInButton")

        // Trigger sign-in flow
        robot.tapElement("LoginSignInButton")
        robot.tapAlertButton("Send Link")

        // WelcomeScreen should appear
        robot.assertElementDisplayed("WelcomeLogo")
        robot.assertElementDisplayed("WelcomeTitle")
        robot.assertElementDisplayed("WelcomeContinueButton")

        // Tap continue to complete flow
        robot.tapElement("WelcomeContinueButton")

        // Flow pops Welcome + Login covers, then navigates to the auth-gated returnTo (Profile)
        robot.assertElementDisplayed("ProfileName")
    }
}

File: e2e-tests/src/main/kotlin/com/mockdonalds/app/e2e/suites/GuestJourneyTest.kt
package com.mockdonalds.app.e2e.suites

import androidx.test.ext.junit.runners.AndroidJUnit4
import com.mockdonalds.app.e2e.robots.AppRobot
import com.mockdonalds.app.features.home.api.ui.HomeTestTags
import com.mockdonalds.app.features.login.api.ui.LoginTestTags
import com.mockdonalds.app.features.more.api.ui.MoreTestTags
import com.mockdonalds.app.features.order.api.ui.OrderTestTags
import com.mockdonalds.app.features.rewards.api.ui.RewardsTestTags
import com.mockdonalds.app.features.scan.api.ui.ScanTestTags
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Guest user journey — browse the app without authenticating.
 * Verifies core screens render, tab navigation works, and
 * auth-gated features redirect to login.
 */
@RunWith(AndroidJUnit4::class)
class GuestJourneyTest {

    private val robot = AppRobot()

    @Before
    fun setUp() {
        robot.launchApp()
    }

    @Test
    fun homeScreenRendersOnLaunch() {
        robot.assertElementDisplayed(HomeTestTags.USER_NAME)
        robot.assertElementDisplayed(HomeTestTags.HERO_BANNER)
    }

    @Test
    fun browseAllTabs() {
        // Home is default
        robot.assertElementDisplayed(HomeTestTags.USER_NAME)

        // Order tab
        robot.tapTab("Order")
        robot.assertElementDisplayed(OrderTestTags.FEATURED_ITEMS_SECTION)

        // Rewards tab
        robot.tapTab("Rewards")
        robot.assertElementDisplayed(RewardsTestTags.POINTS_SECTION)

        // Scan tab
        robot.tapTab("Scan")
        robot.assertElementDisplayed(ScanTestTags.SCAN_BUTTON)

        // More tab
        robot.tapTab("More")
        robot.assertElementDisplayed(MoreTestTags.MENU_LIST)

        // Back to home
        robot.tapTab("Home")
        robot.assertElementDisplayed(HomeTestTags.USER_NAME)
    }

    @Test
    fun profileNavigationRedirectsToLoginWhenUnauthenticated() {
        robot.tapTab("More")
        robot.assertElementDisplayed(MoreTestTags.PROFILE_SECTION)
        robot.tapElement(MoreTestTags.PROFILE_SECTION)

        // Auth interception should redirect to login
        robot.assertElementDisplayed(LoginTestTags.BRANDING)
        robot.assertElementDisplayed(LoginTestTags.SIGN_IN_BUTTON)
    }
}
